sources/core/Xenko.Core.Mathematics/Half2.cs

```diff
@@ -175,13 +175,33 @@ public override bool Equals(object obj)
         }
 
         /// <summary>
-        /// Performs an explicit conversion from <see cref="Vector3"/> to <see cref="Half3"/>.
+        /// Performs an explicit conversion from <see cref="Half2"/> to <see cref="Vector2"/>.
         /// </summary>
         /// <param name="value">The value.</param>
         /// <returns>The result of the conversion.</returns>
         public static explicit operator Vector2(Half2 value)
         {
             return new Vector2(value.X, value.Y);
         }
+
+        /// <summary>
+        /// Performs an explicit conversion from <see cref="Double2"/> to <see cref="Half2"/>.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <returns>The result of the conversion.</returns>
+        public static explicit operator Half2(Double2 value)
+        {
+            return new Half2((Half)value.X, (Half)value.Y);
+        }
+
+        /// <summary>
+        /// Performs an explicit conversion from <see cref="Half2"/> to <see cref="Double2"/>.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <returns>The result of the conversion.</returns>
+        public static explicit operator Double2(Half2 value)
+        {
+            return new Double2(value.X, value.Y);
+        }
```

Comment:
Let's move over all those new implicit operators from `HalfX`/`VectorX` files to `DoubleX` files so that `DoubleX` files are self contained.

Reply:
Alright, so now, `DoubleX.cs` depends on `VectorX` & `HalfX`, but not the other way around.